### PR TITLE
Fix TabControl's updated switching-on-removal logic not considering IsSwitchable

### PR DESCRIPTION
**Pull request by peppy**
Allow's osu!'s `ChannelTabControl` to work again. Without this check, it was possible that the switch call would re-select the tab to be removed.

### CODE REVIEW

File: osu.Framework/Graphics/UserInterface/TabControl.cs [hunk 1]
```diff
@@ -251,12 +256,13 @@ protected virtual void RemoveTabItem(TabItem<T> tab, bool removeFromDropdown = t
...
             {
-                if (Items.Count == 1)
+                if (SwitchableTabs.Length < 2)
                     SelectedTab = null;
                 else
                 {
-                    tabMap.TryGetValue(Items.Last(), out TabItem<T> lastTab);
-                    SwitchTab(tab == lastTab ? -1 : 1);
+                    // check all tabs as to include self (in correct iteration order)
+                    bool anySwitchableTabsToRight = tabMap.Values.SkipWhile(t => t != tab).Skip(1).Any(t => t.IsSwitchable);
```
Comment (smoogipoo) [inline osu.Framework/Graphics/UserInterface/TabControl.cs:264]: The order of `.Values` [isn't specified](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.values?view=netcore-3.1).

I also don't like that neither this code nor the original considers the order of the tabs, as defined by `TabContainer.AllTabItems` (see previous implementation of `SwitchTab()`).

File: osu.Framework/Graphics/UserInterface/TabControl.cs [hunk 2]
```diff
@@ -312,31 +318,20 @@ public virtual void SwitchTab(int direction, bool wrap = true)
             if (Math.Abs(direction) != 1)
                 throw new ArgumentException("value must be -1 or 1", nameof(direction));
 
-            TabItem<T>[] switchableTabs = TabContainer.AllTabItems.Where(tab => tab.IsSwitchable).ToArray();
-            int tabCount = switchableTabs.Length;
+            // the current selected tab may be an non-switchable tab, so search all tabs for a candidate.
+            // this is done to ensure ordering (ie. if an non-switchable tab is in the middle).
+            var allTabs = tabMap.Values.Where(t => t.IsSwitchable || t == SelectedTab);
```
Comment (smoogipoo) [inline osu.Framework/Graphics/UserInterface/TabControl.cs:323]: Same here - this should use the order of appearance as the original code did. The fact this doesn't cause a test failure is concerning, to say the least.

Comment (peppy) [reply osu.Framework/Graphics/UserInterface/TabControl.cs:323]: The fact that I can't understand that there even *is* an order is more concerning. There's too many variables and too many ways to access tabs.

Comment (smoogipoo) [reply osu.Framework/Graphics/UserInterface/TabControl.cs:323]: Yeah, agreed. I missed it in the previous changes too. We did discuss removing it, but nothing came of it.

Comment (peppy) [reply osu.Framework/Graphics/UserInterface/TabControl.cs]: Added a comment about this, should be resolved now.

**Pull request merged**